cider/exec/module/CiderRuntimeModule.cpp

```diff
@@ -744,14 +744,32 @@ CiderBatch CiderRuntimeModule::setSchemaAndUpdateCountDistinctResIfNeed(
       int flatten_index = schema->getFlattenColIndex(i);
       if (schema->getColHints()[i] == ColumnHint::PartialAVG &&
           target_infos[flatten_index].agg_kind == SQLAgg::kSUM &&
-          target_infos[flatten_index].agg_arg_type.is_integer() &&
           generator::getSQLTypeInfo(schema->getColumnTypeById(i).struct_().types(0))
                   .get_type() == SQLTypes::kDOUBLE) {
-        int64_t* result = const_cast<int64_t*>(
-            reinterpret_cast<const int64_t*>(outBuffers[flatten_index]));
-        double value = (double)result[0];
-        double* cast_buffer = reinterpret_cast<double*>(result);
-        cast_buffer[0] = value;
+        if (target_infos[flatten_index].agg_arg_type.is_integer()) {
+          int64_t* result = const_cast<int64_t*>(
+              reinterpret_cast<const int64_t*>(outBuffers[flatten_index]));
+          double* cast_buffer = reinterpret_cast<double*>(result);
+          switch (result[0]) {
+            case std::numeric_limits<int8_t>::min():
+            case std::numeric_limits<int16_t>::min():
+            case std::numeric_limits<int32_t>::min():
+            case std::numeric_limits<int64_t>::min():
```

Comment:
If `agg_arg_type.is_integer()`, the null value could only be null of `int64_t` and no other cases, since `sum(integer)->bigint`.

Reply:
What if we got a substrait plan like in substrait-java keeps the same column type for sum output col as input col which doesn't meet our requirement? I think doing this will enhance our compatibility.

Reply:
It's limited by Cider rule:
`    case SQLAgg::kSUM:
      return arg_expr->get_type_info().is_integer()
                 ? SQLTypeInfo(SQLTypes::kBIGINT, false)
                 : arg_expr->get_type_info();`
So here in the memory, there is no chance of tinyint/smallint/int/float null value exist.
For the case you are talking about, you should update `extract_column` for different null handle of different type, similar as float. And I am not sure whether this problem exists.